### PR TITLE
Readme!

### PR DESCRIPTION
**Pull request by ad2969**
# bt-web
Biztech's web application.

## Getting Started

These instructions will get you a copy of the project up and running on your local machine for development and testing purposes. See [deployment](#deployment) for notes on how to deploy the project to a live system.

### Installation and Running

1. Clone the repo:

    ```
    $ git clone https://github.com/ubc-biztech/bt-web
    ```

2. Install the packages:

    ```
    $ npm install
    ```

3. Your setup is ready for development! Run `npm start` to run your app on your local machine.

### Files and Services

* `components/` - Our main view/display components
* `pages/` - Our main pages, where most of our app logic is handled

* `assets/` - Images, fonts, vectors, and other static files
* `constants/` - Abstracted constants
* `utils/` - Any other miscellaneous functions

* `actions/` - Action creators for redux
* `reducers/` - Reducers (action listeners) for redux

### Linting

Linters are put into place to exercise good and consistent coding style, regardless of developer. Editing lint rules can be done by changing the `.eslintrc.js` file.

## Deployment

Our web application can be deployed through a simple build-and-deploy process:

### Build
The web application can be built using either of the commands:
```
npm run generate
```
or for production environment:
```
npm run generate:prod
```

## Available Scripts

In the project directory, you can run:

```
npm start
```
Runs the app in the development mode. Open [http://localhost:3000](http://localhost:3000) to view it in the browser.

The page will reload if you make edits, and will log out any errors in the console.

```
npm run build
```
Builds the app for production to the `build` folder.
It correctly bundles React in production mode and optimizes the build for the best performance.

See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.

```
npm run generate
```
Builds the app into the `build` folder for stage environment.

```
npm run generate:prod
```
Deploys the app from the `build` folder for production use.

```
npm run deploy
```
Deploys the app from the `build` folder to aws s3 for website hosting.

```
npm run test
```
Launches the test runner in the interactive watch mode.
See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.

```
npm run lint
```
Runs linter and automatically fixes all lint problems.
You may need to install eslint first `npm i -g eslint`

## Contributing
Contributions are accepted from members of the biztech team. General instructions to start contributing are as follows:

1. Clone the remote repo into a local environment
2. Setup the repo (instructions [here](#getting-started))
3. Make the appropriate edits and additions in your own new branch (use a unique branch name!)
4. Write or make changes to any tests, if required
5. Submit pull requests with a detailed description of the modifications
--> Pull requests will be accepted after being reviewed and after the appropriate tests are conducted
6. Merging to master will deploy the API to **staging** environment
7. Only deploy to our **production** environment after fully testing on staging

### Built With

* [React](https://reactjs.org/) - Javascript library

### CODE REVIEW

**Comment by dchen150** [comment]
can you add a line in the Installation and Running about copying over the keys from .env (but don't actually include the keys anywhere)? otherwise LGTM!

File: README.md [hunk 1]
```diff
@@ -1,37 +1,119 @@
...
+## Development
+
+### Files and Services
+
+* `components/` - Our main view/display components
+* `pages/` - Our main pages, where most of our app logic is handled
+
+* `assets/` - Images, fonts, vectors, and other static files
+* `constants/` - Abstracted constants
+* `utils/` - Any other miscellaneous functions
```
Comment (jacqueschen1) [inline README.md:34]: Functions to call our backend API, AWS services, and any other miscellaneous functions

File: README.md [hunk 2]
```diff
@@ -1,37 +1,119 @@
...
+3. Make the appropriate edits and additions in your own new branch (use a unique branch name!)
+4. Write or make changes to any tests, if required
+5. Submit pull requests with a detailed description of the modifications
+--> Pull requests will be accepted after being reviewed and after the appropriate tests are conducted
+6. Merging to master will deploy the API to **staging** environment
+7. Only deploy to our **production** environment after fully testing on staging
+
+### Built With
+
+* [React](https://reactjs.org/) - Javascript library
```
Comment (jacqueschen1) [inline README.md:119]: Haha only react here? :P

**Review by jacqueschen1 — APPROVED** [review]

**Comment by ad2969** [comment]
@dchen150 I forgot one thing as well --  I never really understood how deployment for bt-web works, this was something I thought you could shine a light on better haha

File: README.md (hunk 2 above)
Comment (ad2969) [reply README.md]: calling out the lazy in me 😭

**Comment by dchen150** [comment]
> @dchen150 I forgot one thing as well -- I never really understood how deployment for bt-web works, this was something I thought you could shine a light on better haha -- esp. if we have different environments?
> 
> ps. I also ended up uninstalling a few packages that were apparently not really used in the app?

@ianmah  might know best 😓 , i don't really know the exact difference between dev/staging for us, but i think when you merge to master it gets deployed to prod?

**Comment by ianmah** [comment]
@dchen150 what's the question?

**Comment by dchen150** [comment]
> @dchen150 what's the question?

>I never really understood how deployment for bt-web works @ianmah

**Comment by ianmah** [comment]
Oh shoot i misread your earlier message.

If you take a look at our deployment pipeline, you can see the steps. https://github.com/ubc-biztech/bt-web/blob/d6f927624c58821a18199c706714986bc2f4a9e0/.travis.yml#L24-L25

When a commit is pushed to master, it will trigger a deploy. The deploy steps are also in travis, but basically it builds the production files using `npm run generate:prod`. You can inspect that script in `package.json` => that command runs `cross-env GENERATE_SOURCEMAP=false REACT_APP_STAGE=production npm run build`, which basically sets the stage variable to PROD so that we use the correct deployment configurations in places like this: https://github.com/ubc-biztech/bt-web/blob/d6f927624c58821a18199c706714986bc2f4a9e0/src/utils/index.js#L12-L14

Once the build files are generated, we deploy that to firebase. Read more about that here https://docs.travis-ci.com/user/deployment/firebase/

**Comment by ianmah** [comment]
There's no staging

**Comment by jacqueschen1** [comment]
> Oh shoot i misread your earlier message.
> 
> If you take a look at our deployment pipeline, you can see the steps.
> 
> https://github.com/ubc-biztech/bt-web/blob/d6f927624c58821a18199c706714986bc2f4a9e0/.travis.yml#L24-L25
> 
> When a commit is pushed to master, it will trigger a deploy. The deploy steps are also in travis, but basically it builds the production files using `npm run generate:prod`. You can inspect that script in `package.json` => that command runs `cross-env GENERATE_SOURCEMAP=false REACT_APP_STAGE=production npm run build`, which basically sets the stage variable to PROD so that we use the correct deployment configurations in places like this:
> 
> https://github.com/ubc-biztech/bt-web/blob/d6f927624c58821a18199c706714986bc2f4a9e0/src/utils/index.js#L12-L14
> 
> Once the build files are generated, we deploy that to firebase. Read more about that here https://docs.travis-ci.com/user/deployment/firebase/

It seems that both prod and dev environments use the same cognito pools. Is that true?

**Review by dchen150 — APPROVED** [review]
lgtm

**Pull request merged**